test-runner/wasi_test_runner/filters.py

```diff
@@ -0,0 +1,24 @@
+from typing import Tuple, Any
+from abc import ABC
+from abc import abstractmethod
+
+import json
+
+
+class TestFilter(ABC):
+    @abstractmethod
+    def should_skip(self, test_suite_name: str, test_name: str) -> Tuple[bool, Any]:
+        pass
+
+
+class JSONTestExcludeFilter(TestFilter):
+    def __init__(self, filename: str) -> None:
+        with open(filename, encoding="utf-8") as file:
+            self.filter_dict = json.load(file)
+
+    def should_skip(self, test_suite_name: str, test_name: str) -> Tuple[bool, Any]:
```

Comment:
Should that be `str` instead of `Any`?

Reply:
i made it Any because it can be None.

Reply:
In that case it should be `Tuple[bool, Optional[str]]`

Reply:
iirc such complex types are rejected by tools i was using when writing this code. i guess it's better to avoid being too eager for now.

Reply:
What are those tools? I think `mypy` should be able to handle that. Unless we really except passing here something other than `str` or `None`, I'd rather stick to precise types. That also helps understanding the code; in fact, I wasn't sure what is being returned here until I've looked to the implementation.

Reply:
We already have a few static code analyzers as part of CI (including mypy); I'd suggest giving it a try, and if for some reason it fails, we can open an issue for mypy and switch back to `Any` for now.

Reply:
i tried more strict annotation and for some reasons it worked today.

maybe the annotation which didn't work was different from this version. i don't remember.